presto-testing/src/main/java/io/prestosql/testing/TestingPrestoClient.java

```diff
@@ -224,7 +230,23 @@ else if (TIME_WITH_TIME_ZONE.equals(type)) {
             }
         }
         else if (type instanceof TimestampType) {
-            return SqlTimestamp.JSON_FORMATTER.parse((String) value, LocalDateTime::from);
+            Matcher matcher = TIMESTAMP_PATTERN.matcher((String) value);
+            if (!matcher.matches()) {
+                throw new IllegalArgumentException("Invalid timestamp: " + value);
+            }
+
+            int year = parseInt(matcher.group("year"));
+            int month = parseInt(matcher.group("month"));
+            int day = parseInt(matcher.group("day"));
+            int hour = parseInt(matcher.group("hour"));
+            int minute = parseInt(matcher.group("minute"));
+            int second = parseInt(matcher.group("second"));
+            String fraction = matcher.group("fraction");
+            fraction += "0".repeat(9 - fraction.length());
+            int fractionValue = parseInt(fraction);
+
```

Comment:
This needs to round to nanoseconds if precision is > 9. See VarcharToTimestampCast.castToLongTimestamp